src/config/backend/command_line_arguments.cpp

```diff
@@ -0,0 +1,149 @@
+#include <boost/algorithm/string/join.hpp>
+#include <boost/algorithm/string/split.hpp>
+
+#include "silo/common/panic.h"
+#include "silo/config/util/config_exception.h"
+
+namespace silo::config {
+
+std::string CommandLineArguments::configKeyPathToString(const ConfigKeyPath& key_path) {
+   std::vector<std::string> result;
+   for (const auto& sublevel : key_path.getPath()) {
+      for (const std::string& current_string : sublevel) {
+         result.push_back(current_string);
+      }
+   }
+   return "--" + boost::join(result, "-");
+}
+
+AmbiguousConfigKeyPath CommandLineArguments::stringToConfigKeyPath(
+   const std::string& command_line_argument
+) {
+   if (command_line_argument.size() < 3 || !command_line_argument.starts_with("--")) {
+      throw silo::config::ConfigException(fmt::format(
+         "the provided option '{}' is not a valid command line option", command_line_argument
+      ));
+   }
+
+   std::vector<std::string> delimited_strings;
+   // Remove the leading dashes and split by '-'
+   std::string trimmed = command_line_argument.substr(2);
+   std::vector<std::string> tokens;
+
+   boost::split(tokens, trimmed, boost::is_any_of("-"));
+
+   for (const auto& token : tokens) {
+      if (token.empty()) {
+         throw silo::config::ConfigException(fmt::format(
+            "the provided option '{}' is not a valid command line option", command_line_argument
+         ));
+      }
+      delimited_strings.push_back(token);
+   }
+
+   auto result = AmbiguousConfigKeyPath::tryFrom(std::move(delimited_strings));
+   if (result == std::nullopt) {
+      throw silo::config::ConfigException(fmt::format(
+         "the provided option '{}' is not a valid command line option", command_line_argument
+      ));
+   }
+   return result.value();
+}
+
+namespace {
+
+std::optional<std::string> tryGetAt(const std::vector<std::string>& args, size_t index) {
+   if (index < args.size()) {
+      return args[index];
+   }
+   return std::nullopt;
+}
+
+class ValueAndConsumedFlag {
+  public:
+   ConfigValue value;
+   bool consumed_next;
+};
+
+ValueAndConsumedFlag getValueFromArg(
+   const ConfigValueSpecification& value_specification,
+   const std::string& arg,
+   const std::optional<std::string>& next_arg
+) {
+   if (value_specification.type == ConfigValueType::BOOL) {
+      return {ConfigValue::fromBool(true), false};
+   }
+   if (!next_arg.has_value()) {
+      // VerificationError::ParseError in Rust
+      throw silo::config::ConfigException("missing argument after option " + arg);
+   }
+   return {value_specification.getValueFromString(next_arg.value()), true};
+}
+
+}  // namespace
+
+VerifiedConfigSource CommandLineArguments::verify(const ConfigSpecification& config_specification
+) const {
+   // Parse the command line, now that we have the option keys
+   // and the info about whether they take an argument (any that
+   // are not of type bool).
+
+   // E.g. "--api-foo" => "1234"
```

Comment:
IMO this comment provides very little benefit beside obvious context.
```suggestion
```

Reply:
I've reworded the first comment (it clarifies why the parsing only happens in verify, which actually tripped up Alex). The second comment is now ``// E.g. "--api-foo" => "1234" or "--api-foo=1234"`, and I feel that's useful, makes reading quicker by giving an example instead of having to reconstruct an image in the head. Do you not agree?